src-ext/TypeScript/ggvega/src/itmLayerArrayByScales.ts

```diff
@@ -0,0 +1,37 @@
+import * as GG from '../../ggschema/src/index';
+import {ItmLayer} from './itmLayer';
+
+/**
+ * Modify a layer array according to a scale array
+ *
+ * I think the match-and-delete strategy for scales can be very similar
+ * to the one used in {@link itmLayerArrayByLabelsObject}.
+ *
+ * The obvious difference is that scales are in array, rather than in an object.
+ * Another difference is that a scale has an element called `aesthetics` that
+ * can make it simpler to match to the encoding key. Perhaps {@link contains}
+ * could be used.
+ *
+ * **Called by**
+ * @see layerArray
+ *
+ * @param itmLayerArray
+ * @param ggScaleArray
+ *
+ * @returns `ItmLayer[]`
+ */
+export function itmLayerArrayByScalesArray(itmLayerArray: ItmLayer[], ggScaleArray: GG.Scale[]): ItmLayer[] {
+  // suspect we will need metadata
```

Comment:
Even if it does not do anything yet, as we are working on scales, can you set up some loops here?

The code does not have to return anything yet, but it will be useful to start thinking about the logic.

Reply:
done